vumi/message.py

```diff
@@ -27,12 +27,17 @@ def format_vumi_date(timestamp):
 def parse_vumi_date(value):
     """Parse a timestamp string using the Vumi date format.
 
+    Timestamps without microseconds are also parsed correctly.
+
     :param str value:
         The string to parse.
     :return datetime:
         A datetime object representing the timestamp.
     """
-    return datetime.strptime(value, VUMI_DATE_FORMAT)
+    date_format = VUMI_DATE_FORMAT
+    if "." not in value:
+        date_format = date_format.split(".")[0]
```

Comment:
We could check the last ten characters or so instead of the whole string if you'd prefer. A quick test on my system indicates that scanning a 50MB string for a "." is about an order of magnitude cheaper than parsing that 50MB string out of JSON, though.